src/stan/lang/generator/generate_write_array_method.hpp

```diff
@@ -75,39 +65,80 @@ namespace stan {
         << EOL2;
 
       generate_try(2, o);
-      generate_local_var_decls(prog.derived_decl_.first, 3, o);
-      o << EOL;
-      generate_statements(prog.derived_decl_.second, 3, o);
-      o << EOL;
-
-      generate_comment("validate transformed parameters", 3, o);
-      generate_validate_var_decls(prog.derived_decl_.first, 3, o);
-      o << EOL;
+      if (prog.derived_decl_.first.size() > 0) {
+        generate_comment("declare and define transformed parameters", 2, o);
+        for (size_t i = 0; i < prog.derived_decl_.first.size(); ++i) {
+          generate_indent(3, o);
+          o << "current_statement_begin__ = "
+            <<  prog.derived_decl_.first[i].begin_line_ << ";"
+            << EOL;
+          generate_block_var(prog.derived_decl_.first[i], "double", 3, o);
+          o << EOL;
+        }
+      }
+
+      if (prog.derived_decl_.second.size() > 0) {
+        generate_comment("do transformed parameters statements", 3, o);
+        generate_statements(prog.derived_decl_.second, 3, o);
+        o << EOL;
+      }
+
+      if (prog.derived_decl_.first.size() > 0) {
+        generate_comment("validate transformed parameters", 3, o);
+        o << INDENT3
+          << "const char* function__ = \"validate transformed params\";"
```

Comment:
I think I already suggested pulling this and the void to suppress warnings out into an encapsulated function call.